archive_georep_changelogs.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+import sys
+import xattr
+import struct
+import os
+import shutil
+import urllib.request, urllib.parse, urllib.error
+from pathlib import Path
+import configparser
+import tarfile
+import time
+
+GLUSTERFIND_DIR = "/var/lib/glusterd/glusterfind"
+config = configparser.ConfigParser()
+
+
+def get_glusterfind_time(brick_path):
+    status_file = "{0}.status".format(urllib.parse.quote_plus(brick_path))
+    stimes = []
+
+    # Each session Dir
+    for session in os.listdir(GLUSTERFIND_DIR):
+        full_path = os.path.join(GLUSTERFIND_DIR, session)
+        if os.path.isdir(full_path):
+            # Each volume directory in session dir
+            for v in os.listdir(full_path):
+                full_path_2 = os.path.join(full_path, v)
+                if os.path.isdir(full_path_2):
+                    # Each file in session dir and match for brick status file
+                    for f in os.listdir(full_path_2):
+                        if f == status_file:
+                            with open(os.path.join(full_path_2, f)) as sf:
+                                stimes.append(int(sf.read()))
+
+    if stimes:
+        return min(stimes)
+
+
+def get_archive_stime(brick_path):
+    """
+    Get all stime xattrs and find minimum stime
+    """
+    """
+    When removing georep sessions dangling stimes will remain with a value of "0"
+    We want to avoid them , so we check for the stime-xattr-prefix in each gsyncd.conf
```

Comment:
How about ignoring the xattr values with value `"0"` and use the previous method of xattr list?

```python
if stime[0] > 0:
    stime_xattr_values.append(stime[0])
```

Reply:
Never thought about such scenario. It definitely needs better logic.